src/midisequencer.cpp

```diff
@@ -246,6 +263,33 @@ namespace midisequencer{
         }
     }
 
+    std::chrono::microseconds sequencer::get_start_skipping_silence() {
+        for (auto& msg: messages) {
+            // If we find Loop Start before the first NoteOn, just start there
+            if (is_loop_start(msg.message)) {
+                std::chrono::microseconds time = msg.time;
+                // RPG_RT always rewinds "a little"
+                // This amount is based on the tempo, and this 2100000 divisor
+                // I determined experimentally.
+                time = std::max(0us, std::chrono::microseconds(static_cast<int>(time.count() - (msg.tempo / 2.1f))));
+                return time;
+            } else if ((msg.message & 0xFF) == 0xF0) {
+                // SysEx message. RPG_RT doesn't skip silence if there's a SysEx
+                // message in the beginning, so neither should we...
+                return 0us;
```

Comment:
When it says in the beginning, is from 1:1:000? What if the SysEx message exists at the 1st second and the song begins with notes at the 2nd second? will it pause 1 or 2 seconds?

Reply:
It won't pause at all. If there's any SysExs before the first NoteOn it'll give up. Since it's been a while I can't remember if I definitively verified this behaviour against RPG_RT but this is my understanding of how it behaves. I recall from my youth that harmony would aggressively abandon attempts to skip silence when it encounters any SysExs before any notes play.